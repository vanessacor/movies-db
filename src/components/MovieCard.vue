<template>
	<article :key="movie.title" class="movie-card">
		<router-link :to="{name: 'movieDetails', params: {id: movie.id}}" class="movie-card-link">
			<img :src="movie.poster" alt="" class="movie-img" />
			<p>{{ movie.title }}</p>
		</router-link>
	</article>
</template>

<script>
	export default {
		name: "movieCard",
		props: {
			movie: Object,
		},
	};
</script>

<style scope lang="scss">
	.movie-card {
		max-width: 200px;
		margin-bottom: 20px;
		margin-right: 20px;
	}
	a {
		color: $base-color-light;
		text-decoration: none;
	}

	a:hover {
		color: $highlight-color;
	}

	a:hover img {
		filter: grayscale(100%);
	}
	.movie-img {
		width: 200px;
		height: 300px;
	}
</style>
