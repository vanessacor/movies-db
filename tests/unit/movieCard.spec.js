import { RouterLinkStub, shallowMount } from "@vue/test-utils";

import MovieCard from "@/components/MovieCard";

describe("test Movie Card", () => {
  const wrapper = shallowMount(MovieCard, {
    stubs: {
      RouterLink: RouterLinkStub,
    },

    propsData: {
      movie: {
        title: "Mock movie",
        director: "Vanessa",
        description:
          "lorem Ipsum lorem Ipsumlorem Ipsum lorem Ipsum lorem Ipsum lorem Ipsumlorem Ipsum lorem Ipsumlorem Ipsum lorem Ipsum",
        poster: "https://somewhere.com.somepic.jpg",
        duration: "133",
        rating: "5",
        genre: "Mock",
        year: "1997",
        id: 9,
      },
    },
  });
  it("should render movie title", () => {
    expect(wrapper.text()).toContain("Mock movie");
  });
  it("should render img", () => {
    const image = wrapper.find("img");
    expect(image.exists()).toBe(true);
  });
  it("should render the routerLink", () => {
    const link = wrapper.find(RouterLinkStub);
    expect(link.props().to.name).toEqual("movieDetails");
    expect(link.props().to.params).toEqual({ id: 9 });
  });
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
