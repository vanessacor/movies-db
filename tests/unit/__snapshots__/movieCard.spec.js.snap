// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Movie Card should match the snapshot 1`] = `
<article class="movie-card"><a class="movie-card-link"><img src="https://somewhere.com.somepic.jpg" alt="" class="movie-img">
    <p>Mock movie</p>
  </a></article>
`;
